### PR TITLE
Sharing: completes one of two needed code merges to bring the code in sync with wpcom.

### PR DESCRIPTION
**Pull request by zinigor**
This branch will let us sync the code from Jetpack back to wpcom when this is merged to `master`.

### CODE REVIEW

File: modules/sharedaddy/sharing.js [hunk 1]
```diff
@@ -20,12 +20,8 @@ if ( sharing_js_options && sharing_js_options.counts ) {
 				}
 
 				// get both the http and https version of these URLs
-				https_url = encodeURIComponent( url.replace( /^http:\/\//i, 'https://' ) );
```
Comment (aduth) [inline modules/sharedaddy/sharing.js]: In 7b2e79d, you had consolidated the `encodeURIComponent` from individual requests to encode as part of the `https_url` and `http_url` assignments. You'll either want to push this back to WPCOM (and add back to [lines 23/24](https://github.com/Automattic/jetpack/pull/2119/files#diff-1ef3c76357b55ce0f40b6c21eab8b4bfR23)), or reinstate the `encodeURIComponent` calls when generating URLs for the `urls` object.

File: modules/sharedaddy/sharing-sources.php [hunk 2]
```diff
@@ -391,9 +391,7 @@ function sharing_twitter_via( $post ) {
 
 		// Strip out anything other than a letter, number, or underscore.
 		// This will prevent the inadvertent inclusion of an extra @, as well as normalizing the handle.
-		$twitter_site_tag_value = preg_replace( '/[^\da-z_]+/i', '', $twitter_site_tag_value );
-
-		return $twitter_site_tag_value;
+		return preg_replace( '/[^\da-z_]+/i', '', $twitter_site_tag_value );
```
Comment (aduth) [inline modules/sharedaddy/sharing-sources.php]: Lines are a bit out of order between here and WPCOM as to when the `jetpack_sharing_twitter_via` filter is applied. I think I prefer what you have here (with the comment above the actual stripping).

File: modules/sharedaddy/sharing.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ if ( sharing_js_options && sharing_js_options.counts ) {
 		twitter_count : {},
 		get_counts : function() {
 			var facebookPostIds = [],
-				https_url, http_url, url, urls, id, service, service_url, path_ending;
```
Comment (aduth) [inline modules/sharedaddy/sharing.js]: `url` shouldn't have been removed here - it's still used below in the `https_url`/`http_url` assignment and in the `for` loop iterating over `WPCOM_sharing_counts`.

**Comment by aduth** [comment]
The WPCOM version of `sharing.js` removes [this line](https://github.com/Automattic/jetpack/blob/018c70a7b81244718546f9c4c6e00a5a5b73e48c/modules/sharedaddy/sharing.js#L74), which is unnecessary in light of the `jsonpCallback`.

**Comment by zinigor** [comment]
@aduth thanks for reviewing and catching all those problems, I have fixed all of them, rebased the branch again, can you please take one final look?

**Comment by aduth** [comment]
This looks good to me :+1:

**Pull request merged**